### PR TITLE
Added commutativity in string/bytes multiplication

### LINKED ISSUE #17506: Should string/bytes multiplication be commutative?
Today we can do:

```chapel
writeln("foo"*3);
writeln(b"foo"*3);
```

but we cannot do

```chapel
writeln(3*"foo");
writeln(3*b"foo");  // I was curious whether this would parse, it does (but doesn't resolve)
```

because we don't have the overloads of these operator with symmetric arguments.

I think we should support `int*string` and `int*bytes` operations, as well.

### PR DESCRIPTION
**Pull request by shubhamkmr04**
This PR adds commutativity in bytes and string multiplication with int. 
Resolve #17506

### CODE REVIEW

**Comment by shubhamkmr04** [comment]
@e-kayrakli please tell me if this PR needs any more changes.

File: modules/internal/Bytes.chpl [hunk 1]
```diff
@@ -1097,11 +1097,29 @@ module Bytes {
...
+     Results in::
+
+        Hello! Hello! Hello!         
   */
   operator bytes.*(s: bytes, n: integral) {
     return doMultiply(s, n);
   }
 
+  pragma "no doc"
+  operator bytes.*(n: integral, s: bytes) {
```
Comment (e-kayrakli) [inline modules/internal/Bytes.chpl:1119]: For now, let's convert this and the one above to standalone functions, so they look like:

```
operator *(....)
```

Also pinging @lydia-duncan. The string*integral operator is standalone, but bytes*integral is a method on `bytes`. I think we should make string/bytes consistent in this matter, and if I were to choose, I'd make them standalone with the philosophical argument that all string/bytes/int are built-in types, and as such, operators between them doesn't necessarily belong to one type.

Comment (shubhamkmr04) [reply modules/internal/Bytes.chpl:1119]: okay @e-kayrakli , I understood, will commit the changes soon

**Comment by e-kayrakli** [comment]
Test status:
- [x] standard `types/{string,bytes}`
- [ ] standard full suite

**Comment by shubhamkmr04** [comment]
Hey @e-kayrakli , sorry for replying late. My university exams were going on.

**Comment by shubhamkmr04** [comment]
I am facing a problem. I did the changes in my local, it passed the tests. I pushed the commit to this branch but I don't know how the commit is not reflecting in my PR.

**Comment by e-kayrakli** [comment]
That's strange. Is that commit empty? Did you maybe put that commit on a different branch first? When you are on your branch, how many commits do you see when you do `git log $(git merge-base master HEAD)..HEAD`?

You can always rebase your branch on master to clean up any issue. Something like:

```
git checkout master
git pull upstream master
git checkout Issue#17506
git rebase master  # alternatively throw in `-i` and make sure you'll get a list of 6 commits, not 5, save and quit the file as-is
```

**Comment by shubhamkmr04** [comment]
The commit was not empty, I am still getting 6 commits on `git log` but not in this PR. I tried to rebase the branch as you mentioned above but I think I did some mistake, It's getting really confusing. 
Should I open a different PR? I would do the exact 6 commits in that PR. I guess that would be straight up ready to merge as I passed the tests in my last commits in `test/release/examples/primers/learnChapelInYMinutes.chpl`

**Comment by e-kayrakli** [comment]
It was probably a github hiccup. I think I can see that commit now?

**Comment by shubhamkmr04** [comment]
> It was probably a github hiccup. I think I can see that commit now?

Yes @e-kayrakli , the commit is visible now:) Please have a look at the changes and let me know if I need to do some more changes.

**Comment by e-kayrakli** [comment]
OK. I'm hoping to take another look at this and run some testing later today. Thanks!

**Comment by e-kayrakli** [comment]
@shubhamkmr04 -- My apologies. I couldn't get to take a look at this in time. I will be off for the rest of the week. This is still in my radar, but will need to wait until next week.

**Comment by shubhamkmr04** [comment]
> @shubhamkmr04 -- My apologies. I couldn't get to take a look at this in time. I will be off for the rest of the week. This is still in my radar, but will need to wait until next week.

Sure @e-kayrakli , please have a look at this whenever you get time. Thanks

**Comment by e-kayrakli** [comment]
Thanks @shubhamkmr04 !!

**Comment by shubhamkmr04** [comment]
My pleasure! Finally, I got my 4th PR merged for Chapel. Thanks to you @e-kayrakli for helping me out with this one:)

**Pull request merged**